### PR TITLE
Assorted dragon fight work

### PR DESCRIPTION
**Pull request by shadowfacts**
I'm not sure what the distinction is between PhaseBase and PhaseBaseOther (or why the latter is even necessary), so suggestions are welcome for better names for those.

### CODE REVIEW

File: mappings/net/minecraft/entity/boss/dragon/phase/PhaseBase.mapping [hunk 1]
```diff
@@ -0,0 +1,16 @@
+CLASS ang net/minecraft/entity/boss/dragon/phase/PhaseBase
```
Comment (kashike) [inline mappings/net/minecraft/entity/boss/dragon/phase/PhaseBase.mapping]: `AbstractPhase`

File: mappings/net/minecraft/entity/boss/dragon/phase/PhaseBaseOther.mapping [hunk 2]
```diff
@@ -0,0 +1,7 @@
+CLASS anh net/minecraft/entity/boss/dragon/phase/PhaseBaseOther
```
Comment (kashike) [inline mappings/net/minecraft/entity/boss/dragon/phase/PhaseBaseOther.mapping]: `AbstractSittingPhase`

**Review by kashike — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/entity/boss/dragon/EnderDragonEntity.mapping [hunk 3]
```diff
@@ -17,9 +17,15 @@ CLASS ane net/minecraft/entity/boss/dragon/EnderDragonEntity
 	METHOD U kill ()V
 	METHOD a damage (Laft;F)Z
 		ARG 1 source
+		ARG 2 amount
 	METHOD a damage (Lanc;Laft;F)Z
 		ARG 1 part
 		ARG 2 source
+		ARG 3 amount
+	METHOD a crystalWasDamaged (Land;Let;Laft;)V
```
Comment (modmuss50) [inline mappings/net/minecraft/entity/boss/dragon/EnderDragonEntity.mapping]: possibly drop the was? `crystalDamaged? `

Comment (kashike) [reply mappings/net/minecraft/entity/boss/dragon/EnderDragonEntity.mapping]: Should be `crystalDestroyed` or similar.

File: mappings/net/minecraft/entity/boss/dragon/EnderDragonFight.mapping [hunk 4]
```diff
@@ -1 +1,10 @@
 CLASS brj net/minecraft/entity/boss/dragon/EnderDragonFight
+	FIELD a LOGGER Lorg/apache/logging/log4j/Logger;
+	FIELD e gateways Ljava/util/List;
+	FIELD k dragonKilled Z
+	FIELD l previouslyKilled Z
+	FIELD m dragonUUID Ljava/util/UUID;
```
Comment (asiekierka) [inline mappings/net/minecraft/entity/boss/dragon/EnderDragonFight.mapping]: dragonUUID or dragonUuid?

**Review by asiekierka — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/entity/boss/dragon/EnderDragonFight.mapping (hunk 4 above)
Comment (kashike) [reply mappings/net/minecraft/entity/boss/dragon/EnderDragonFight.mapping]: `dragonUUID`, matches the NBT key.

Comment (asiekierka) [reply mappings/net/minecraft/entity/boss/dragon/EnderDragonFight.mapping]: I thought we agreed not to capitalize UUID (or NBT, or...) in field/method names, though. Which do you think is more important (genuine question) - matching NBT keys or keeping to an unified naming standard?

Comment (kashike) [reply mappings/net/minecraft/entity/boss/dragon/EnderDragonFight.mapping]: I didn't realise that was agreed upon - `dragonUuid` then.

**Review by asiekierka — APPROVED** [review]
I'd say crystalDestroyed might potentially be better as onCrystalDestroyed, please look into that; however it's good enough for merging for sure.

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**